### ISSUE TITLE
Make more flexible

### ISSUE THREAD

**Issue by ultimateshadsform** [romsar/gonertia#24]
Right now isn't it hardcoded to use html/template?

How can I use something else like templ?

[https://templ.guide/](https://templ.guide/)

I did make it working using an adapter but it would be nice to be able to override the template mechanism so I can use templ..

Or can I do this already?

**Comment by romsar** [comment]
I think we can do this. I will try to allocate time to work on this issue in the near future.

**Comment by ultimateshadsform** [comment]
Yeah! Would be nice to be able to override the template thing to use a custom one.

**Comment by romsar** [comment]
I think we can pass something like TemplateProvider interface to the Gonertia constructor like in Fiber: https://docs.gofiber.io/guide/templates/

**Comment by ultimateshadsform** [comment]
I think we can push this into the backlog of 3.x I also would like to be able to use any http framework such as gofiber or like golang http etc. I have cloned and I'm starting to see if I can come up with something

**Comment by ultimateshadsform** [comment]
I also would like to add it would be nice to use as a string too without any html/template stuff either. That would make it more flexible which would allow any developer to implement it however they want. In any template engine etc.

**Comment by ultimateshadsform** [comment]
You can see my example project I edited to fit TypeScript: https://github.com/ultimateshadsform/gonertia-vue
